### ISSUE TITLE
Symfony Level + phpdoc_order leaves docblock not totally fixed.

### ISSUE THREAD

**Issue by cryode** [FriendsOfPHP/PHP-CS-Fixer#1746]
CS Fixer version: 1.11.1

Sample code:

```php
<?php

class Foo
{
    /**
     * @param Exception $exception
     * @param Closure $response
     * @param string $context
     * @return Foo
     * @throws \RuntimeException
     */
    public function handleError(\Exception $exception, \Closure $response, $context)
    {
        // implementation...
    }
}
```

Run only the required phpdoc fixers to achieve the results we want (aligned, ordered, separated, trimmed):

    php-cs-fixer fix example.php --fixers="phpdoc_params,phpdoc_order,phpdoc_separation,phpdoc_trim"

Achieve expected result:

```php
<?php

class Foo
{
    /**
     * @param Exception $exception
     * @param Closure   $response
     * @param string    $context
     *
     * @throws \RuntimeException
     *
     * @return Foo
     */
    public function handleError(\Exception $exception, \Closure $response, $context)
    {
        // implementation...
    }
}
```

However, running the Symfony level (which has all of those except order) plus `phpdoc_order` results in a mis-trim:

    php-cs-fixer fix example.php --level="symfony" --fixers="phpdoc_order"

```php
<?php

class Foo
{
    /**
     * @param Exception $exception
     * @param Closure   $response
     * @param string    $context
     *
     * @throws \RuntimeException
     * @return Foo
     *
     */
    public function handleError(\Exception $exception, \Closure $response, $context)
    {
        // implementation...
    }
}
```

It's correct except for that last line not being trimmed. Running the same command again will fix this as expected and you end up with the desired result.

I can only assume that this is caused either by an order of operations (trim running before something else and not being run again), or some other conflict with an existing Symfony level fixer. I would play with it more to try and figure it out for sure if it weren't the middle of a work day. :)

**Comment by GrahamCampbell** [comment]
The order fixer doesn't run on symfony because symfony use a different order.

**Comment by GrahamCampbell** [comment]
> However, running the Symfony level (which has all of those except order) plus phpdoc_order results in a mis-trim:

Does that line have a trailing space on it by any chance?

**Comment by cryode** [comment]
> The order fixer doesn't run on symfony because symfony use a different order.

I understand why the order fixer isn't in the Symfony level. Just pointing out that all the fixers I'd like to run should be specified by combining the Symfony level with the `phpdoc_order` contrib fixer, yet the result is not the same.

> Does that line have a trailing space on it by any chance?

Does not look like it, no.

**Comment by SpacePossum** [comment]
1.12
```
possum@possum:/home/possum/work/PHP-CS-Fixer$ php-cs-fixer fix test.php --level="symfony" --fixers="phpdoc_order" -vvv
You are running php-cs-fixer with xdebug enabled. This has a major impact on runtime performance.
F
Legend: ?-unknown, I-invalid file syntax, file ignored, .-no changes, F-fixed, E-error
   1) /home/possum/work/PHP-CS-Fixer/test.php (phpdoc_separation, phpdoc_params, phpdoc_order)
Fixing time per file:
[0.073 s] /home/possum/work/PHP-CS-Fixer/test.php
```
Fails

Running with only fixers applied (see above) 1.12 (again)
```
possum@possum:/home/possum/work/PHP-CS-Fixer$ php-cs-fixer fix test.php  --fixers="phpdoc_separation,phpdoc_params,phpdoc_order" -vvv
You are running php-cs-fixer with xdebug enabled. This has a major impact on runtime performance.
F
Legend: ?-unknown, I-invalid file syntax, file ignored, .-no changes, F-fixed, E-error
   1) /home/possum/work/PHP-CS-Fixer/test.php (phpdoc_order, phpdoc_separation, phpdoc_params)
Fixing time per file:
[0.052 s] /home/possum/work/PHP-CS-Fixer/test.php

Fixed all files in 0.052 seconds, 6.000 MB memory used

```
OK

something is odd

>Does that line have a trailing space on it by any chance?

I tested without

**Comment by GrahamCampbell** [comment]
> something is odd

Sounds like the fixer priorities are wrong, but I'm not sure how.

**Comment by GrahamCampbell** [comment]
// cc @keradus

**Comment by cryode** [comment]
I decided to check to see what order they're being run in. After listing all the fixers in order, I isolated the `phpdoc_` prefixed ones and ran only those (including all Symfony + the one contrib):

    php-cs-fixer fix example.php --fixers="phpdoc_to_comment,phpdoc_indent,phpdoc_types,phpdoc_scalar,phpdoc_no_empty_return,phpdoc_no_package,phpdoc_no_access,phpdoc_order,phpdoc_inline_tag,phpdoc_separation,phpdoc_short_description,phpdoc_type_to_var,phpdoc_var_without_name,phpdoc_var_to_type,phpdoc_trim,phpdoc_params"

(The list is in the same order as they should be run, if that helps anyone.)

After running this command, I get the expected result. So somewhere along the lines, one of the other Symfony fixers is interfering. I'll continue to poke around to see if I can narrow it down.

**Comment by cryode** [comment]
And if it helps, here's the full list of Symfony + `phpdoc_order` fixers being run, in order based on priority:

```
encoding: 100
short_tag: 98
indentation: 50
linefeed: 50
unneeded_control_parentheses: 30
phpdoc_to_comment: 25
phpdoc_indent: 20
trailing_spaces: 20
phpdoc_types: 16
phpdoc_scalar: 15
phpdoc_no_empty_return: 10
phpdoc_no_package: 10
phpdoc_no_access: 10
duplicate_semicolon: 10
phpdoc_order: 5
double_arrow_multiline_whitespaces: 1
phpdoc_inline_tag: 0
long_array_syntax: 0
operators_spaces: 0
header_comment: 0
phpdoc_separation: 0
ereg_to_preg: 0
object_operator: 0
no_blank_lines_after_class_opening: 0
join_function: 0
include: 0
list_commas: 0
namespace_no_leading_whitespace: 0
phpdoc_short_description: 0
new_with_braces: 0
no_empty_lines_after_phpdocs: 0
phpdoc_type_to_var: 0
trim_array_spaces: 0
ternary_spaces: 0
standardize_not_equal: 0
unalign_double_arrow: 0
unalign_equals: 0
whitespacy_lines: 0
unary_operators_spaces: 0
spaces_before_semicolon: 0
single_quote: 0
print_to_echo: 0
pre_increment: 0
phpdoc_var_without_name: 0
return: 0
self_accessor: 0
single_blank_line_before_namespace: 0
single_array_no_trailing_comma: 0
function_typehint_space: 0
multiline_array_trailing_comma: 0
lowercase_constants: 0
newline_after_open_tag: 0
lowercase_keywords: 0
php4_constructor: 0
method_argument_space: 0
php_unit_strict: 0
short_array_syntax: 0
function_call_space: 0
function_declaration: 0
no_blank_lines_before_namespace: 0
short_echo_tag: 0
elseif: 0
strict_param: 0
blankline_after_open_tag: 0
array_element_white_space_after_comma: 0
concat_without_spaces: 0
phpdoc_var_to_type: 0
empty_return: 0
visibility: 0
array_element_no_space_before_comma: 0
parenthesis: 0
multiple_use: 0
php_closing_tag: 0
single_line_after_imports: 0
multiline_spaces_before_semicolon: 0
remove_lines_between_uses: -5
phpdoc_trim: -5
short_bool_cast: -9
spaces_cast: -10
psr0: -10
php_unit_construct: -10
unused_use: -10
align_equals: -10
strict: -10
concat_with_spaces: -10
phpdoc_params: -10
logical_not_operators_with_spaces: -10
logical_not_operators_with_successor_space: -10
align_double_arrow: -10
echo_to_print: -10
extra_empty_lines: -20
line_after_namespace: -20
remove_leading_slash_use: -20
braces: -25
ordered_use: -30
eof_ending: -50
```

**Comment by cryode** [comment]
I've confirmed that the priority order is not being adjusted as expected when using this CLI syntax.

I first discovered this by isolating the two configurations giving me problems:

    $ php-cs-fixer fix example.php --level="symfony"
    $ php-cs-fixer fix example.php --fixers="phpdoc_order"

This yielded the same unexpected result as originally demonstrated.

This happens after `ConfigurationResolver::resolve()` runs through [`resolveByLevel()` and `resolveByName()`](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/1.12/Symfony/CS/ConfigurationResolver.php#L130-L178). After the list of fixers is compiled, it is never sorted. This results in having all of the Symfony fixers first, followed by `phpdoc_order` on its own.

I implemented and tested a fix by implementing the same `sortFixers()` method as used in [Symfony\CS\Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/1.12/Symfony/CS/Fixer.php#L337-L342) in the Resolver. A similar fix could be implemented within [`Fixer::prepareFixers()`](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/1.12/Symfony/CS/Fixer.php#L344-L355) as well. The term fixer is a little ambiguous within this project, since it can mean either a list of all fixers, or those to be applied to the current fixing process, so I don't want to make an assumption about which place is best.

With a little direction, I'd be happy to create a pull request. Otherwise, seems like a straight forward fix!
